domain/src/main/java/com/teamagam/gimelgimel/domain/layers/LoadIntermediateRastersInteractor.java

```diff
@@ -0,0 +1,37 @@
+package com.teamagam.gimelgimel.domain.layers;
+
+
+import com.teamagam.gimelgimel.domain.base.executor.ThreadExecutor;
+import com.teamagam.gimelgimel.domain.base.interactors.BaseDataInteractor;
+import com.teamagam.gimelgimel.domain.base.interactors.DataSubscriptionRequest;
+import com.teamagam.gimelgimel.domain.map.repository.IntermediateRastersRepository;
+
+import java.util.Collections;
+
+import javax.inject.Inject;
+
+import rx.Observable;
+
+public class LoadIntermediateRastersInteractor extends BaseDataInteractor {
+
+    private final IntermediateRasterLocalStorage mRastersLocalStorage;
+    private final IntermediateRastersRepository mRastersRepository;
+
+    @Inject
+    public LoadIntermediateRastersInteractor(ThreadExecutor threadExecutor,
+                                             IntermediateRasterLocalStorage rasterLocalStorage,
+                                             IntermediateRastersRepository rastersRepository) {
+        super(threadExecutor);
+        mRastersLocalStorage = rasterLocalStorage;
+        mRastersRepository = rastersRepository;
+    }
+
+    @Override
+    protected Iterable<SubscriptionRequest> buildSubscriptionRequests(DataSubscriptionRequest.SubscriptionRequestFactory factory) {
+        DataSubscriptionRequest subscriptionRequest = factory.create(Observable.just(null)
+                .flatMapIterable(x -> mRastersLocalStorage.getExistingLayers())
```

Comment:
Note to selves: This will need a refactor when rebased on the bug-fix of the subjects issue.
We no longer use this form of an interactor.